packages/@atjson/document/src/collection.ts

```diff
@@ -79,6 +84,28 @@ export class Collection {
       );
     }
 
+    /**
+     * this is just inlining the logic for the common case where the filter object looks like `{ type: '-vendor-type' }` for performance reasons
+     */
+    if (Object.keys(filter).length === 1 && filter.type != null) {
+      let annotations = [];
+      for (let a of this.annotations) {
+        let annotationClass = a.constructor as AnnotationConstructor<
+          typeof a,
+          typeof a.attributes
+        >;
+        let vendorPrefix = annotationClass.vendorPrefix;
+        if (
+          filter.type === `-${vendorPrefix}-${a.type}` ||
+          (a instanceof UnknownAnnotation && a.attributes.type === filter.type)
```

Comment:
I'm seeing a fairly consistent 20% performance improvement in the cumulativeTime of this function by using 
```
(a.type === "unknown" && a.attributes.type === filter.type)
```
here instead of `instanceof`

Reply:
Ahh, maybe we should check using `a.type` and `vendorPrefix`, then?